src/services/getNewsApiAxios.ts

```diff
@@ -0,0 +1,25 @@
+import axios from "axios";
+import { ENDPOINTS } from "../utils/types";
+
+const apiAxios = axios.create({
+  baseURL: `https://newsapi.org/v2/`,
+  timeout: 10000,
+});
+
+export const getArticlesFromApi = async (filtersState: any, location: any) => {
+  //preper url string
+  //send request
+  let country = "";
+  if (filtersState.endpoint === ENDPOINTS.topheadlines) {
+    country = filtersState.country ? filtersState.country : location;
+  }
+  return await apiAxios.get(
+    `${filtersState.endpoint}?q=${filtersState.searchInput}&country=${country}&category=${filtersState.category}&sources=${filtersState.source}&from=${filtersState.dateFrom}&to=${filtersState.dateTo}&language=${filtersState.language}&sortBy=${filtersState.sortBy}&apiKey=${process.env.REACT_APP_API_KEY}`
+  );
```

Comment:
please try to modify the url like in that article: 
https://medium.com/@srph/axios-configure-the-base-path-daed6ff79eab

the idea is to chain the API key to every request

Reply:
I put in a comment axios.interceptor because I dont understand yet how to use it and call it .